sql/create.go

```diff
@@ -440,18 +479,69 @@ func (n *createTableNode) Start() error {
 		}
 	}
 
+	if n.n.IfAsExists {
+		resultColumns := n.selectPlan.Columns()
+		if err != nil {
+			return err
+		}
+		desiredTypesFromSelect := make([]parser.Datum, len(resultColumns))
+		for i, col := range resultColumns {
+			desiredTypesFromSelect[i] = col.Typ
+		}
+		insertPlan, err := n.p.newPlan(&parser.Insert{Table: &n.n.Table, Rows: n.n.Rows}, desiredTypesFromSelect, false)
+		if err != nil {
+			return err
+		}
+		n.insertPlan = insertPlan
+		err = insertPlan.expandPlan()
+		if err != nil {
+			return err
+		}
+		err = insertPlan.Start()
+		if err != nil {
+			return err
+		}
+		for done := true; done; done, err = insertPlan.Next() {
```

Comment:
I have a feeling this loop is more properly placed inside the Next method (where instead of looping we just return insertPlan.Next()) but I don't know how to fire off the cycle for the Next loop for createTableNode (it currently never gets called)